exchanges/order/order_test.go

```diff
@@ -1869,3 +1887,50 @@ func TestDeriveCancel(t *testing.T) {
 		t.Fatalf("unexpected values %+v", cancel)
 	}
 }
+
+func TestGetOrdersRequest_Filter(t *testing.T) {
+	request := new(GetOrdersRequest)
+	request.AssetType = asset.Spot
+	request.Type = AnyType
+	request.Side = AnySide
+
+	var orders = []Detail{
+		{Pair: btcusd, AssetType: asset.Spot, Type: Limit, Side: Buy},
+		{Pair: btcusd, AssetType: asset.Spot, Type: Limit, Side: Sell},
+		{Pair: btcusd, AssetType: asset.Spot, Type: Market, Side: Buy},
+		{Pair: btcusd, AssetType: asset.Spot, Type: Market, Side: Sell},
+		{Pair: btcusd, AssetType: asset.Futures, Type: Limit, Side: Buy},
+		{Pair: btcusd, AssetType: asset.Futures, Type: Limit, Side: Sell},
+		{Pair: btcusd, AssetType: asset.Futures, Type: Market, Side: Buy},
+		{Pair: btcusd, AssetType: asset.Futures, Type: Market, Side: Sell},
+		{Pair: btcltc, AssetType: asset.Spot, Type: Limit, Side: Buy},
+		{Pair: btcltc, AssetType: asset.Spot, Type: Limit, Side: Sell},
+		{Pair: btcltc, AssetType: asset.Spot, Type: Market, Side: Buy},
+		{Pair: btcltc, AssetType: asset.Spot, Type: Market, Side: Sell},
+		{Pair: btcltc, AssetType: asset.Futures, Type: Limit, Side: Buy},
+		{Pair: btcltc, AssetType: asset.Futures, Type: Limit, Side: Sell},
+		{Pair: btcltc, AssetType: asset.Futures, Type: Market, Side: Buy},
+		{Pair: btcltc, AssetType: asset.Futures, Type: Market, Side: Sell},
+	}
+
+	shinyAndClean := request.Filter("test", orders)
+	if len(shinyAndClean) != 16 {
+		t.Fatalf("received: '%v' but expected: '%v'", len(shinyAndClean), 16)
+	}
+
+	request.Pairs = []currency.Pair{btcltc}
+
+	// Kicks off time error
+	request.EndTime = time.Unix(1336, 0)
+	request.StartTime = time.Unix(1337, 0)
+
+	shinyAndClean = request.Filter("test", orders)
+
+	if len(shinyAndClean) != 8 {
+		t.Fatalf("received: '%v' but expected: '%v'", len(shinyAndClean), 8)
+	}
+
+	if !shinyAndClean[0].Pair.Equal(btcltc) {
```

Comment:
I think we could make this a little better. Perhaps a `for` to ensure all returned items are filtered appropriately. Or perhaps orderID checks